nameko_grpc/streams.py

```diff
@@ -166,7 +170,7 @@ def headers_to_send(self, defer_until_data=True):
         if self.headers_sent or len(self.headers) == 0:
             return False
 
-        if defer_until_data and self.queue.empty():
+        if defer_until_data and self.queue.empty() and self.buffer.empty():
```

Comment:
Was this a preexisting bug, or did the implementation need to change as a result of the termination handling?

Reply:
Removed this as It doesn't seem like it was needed and I don't see why I added it.

Reply:
actually I've added it back. 
Previously, we wouldn't send headers until there was data to send (which sat on the queue as it wasn't flushed until the headers had been sent. But now, as a result of calling flush_queue_to_buffer indiscriminately, that data might be on the buffer rather than the queue.
We could possibly remove the check on the queue empty but it feels safer to keep it.